web/src/components/tasks/MultipleDwellingRegistrationTask.tsx

```diff
@@ -0,0 +1,172 @@
+import { Content } from "@/components/Content";
+import { TaskHeader } from "@/components/TaskHeader";
+import { NeedsAccountModalWrapper } from "@/components/auth/NeedsAccountModalWrapper";
+import { PrimaryButton } from "@/components/njwds-extended/PrimaryButton";
+import { SecondaryButton } from "@/components/njwds-extended/SecondaryButton";
+import { CheckHousingRegistrationStatus } from "@/components/tasks/CheckHousingRegistrationStatus";
+import { HousingRegistrationStatusSummary } from "@/components/tasks/HousingRegistrationStatusSummary";
+import { UnlockedBy } from "@/components/tasks/UnlockedBy";
+import { HousingMunicipalitiesContext } from "@/contexts/housingMunicipalitiesContext";
+import * as api from "@/lib/api-client/apiClient";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useRoadmap } from "@/lib/data-hooks/useRoadmap";
+import { MultipleDwellingSearchError, Task } from "@/lib/types/types";
+import { openInNewTab } from "@/lib/utils/helpers";
+import { getModifiedTaskContent } from "@/lib/utils/roadmap-helpers";
+import {
+  HousingAddress,
+  HousingRegistrationRequestLookupResponse,
+} from "@businessnjgovnavigator/shared/housing";
+import { TabContext, TabList, TabPanel } from "@mui/lab/";
+import { Box, Tab } from "@mui/material";
+import React, { ReactElement, useContext, useState } from "react";
+
+interface Props {
+  task: Task;
+  CMS_ONLY_disable_overlay?: boolean;
+}
+
+const APPLICATION_TAB_INDEX = 0;
+const STATUS_TAB_INDEX = 1;
+
+export const MultipleDwellingRegistrationTask = (props: Props): ReactElement => {
+  const { roadmap } = useRoadmap();
+  const callToActionLink = getModifiedTaskContent(roadmap, props.task, "callToActionLink");
+  const [tabIndex, setTabIndex] = useState(APPLICATION_TAB_INDEX);
+  const [error, setError] = useState<MultipleDwellingSearchError | undefined>(undefined);
+  const [housingAddress, setHousingAddress] = useState<HousingAddress | undefined>(undefined);
+  const [housingRegistrationLookupSummary, setHousingRegistrationLookupSummary] = useState<
+    HousingRegistrationRequestLookupResponse | undefined
+  >(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { municipalities } = useContext(HousingMunicipalitiesContext);
+  const { Config } = useConfig();
+
+  const onSelectTab = (event: React.SyntheticEvent, newValue: string): void => {
+    const index = Number.parseInt(newValue);
+    setTabIndex(index);
+  };
+
+  const onEdit = (): void => {
+    setHousingAddress(undefined);
+    setHousingRegistrationLookupSummary(undefined);
+  };
+
+  if (municipalities.length === 0) {
+    setError(`SEARCH_FAILED`);
+  }
```

Comment:
We just went through something very similar with the license search. Based on what I'm seeing, the case we're looking at here displays an error message: "Sorry, something went wrong. Please try again later." This implies that the service has an issue, but it could be that the user fat-fingered some data. Trying again may not solve the issue.

It might be worth having a different error message for a successful search that returns no data.

Reply:
So this would be the service having an error: specifically if we can't pull the list of municipalities, they can't search because of it. It would be unlikely to actually have this happen though, since we're pulling from a static list of municipalities. We do have dedicated errors per:

1. We can't find anything for the address you put in
2. We found a property at that address but no registrations

Reply:
Ah that's interesting. So a slightly different use case. Great to know. thank you!